core/backend/src/IModelDb.ts

```diff
@@ -606,6 +608,32 @@ export abstract class IModelDb extends IModel {
     return stmt;
   }
 
+  /**
+   * queries the BisCore.SubCategory table for the entries that are children of the passed categoryIds
+   * @param categoryIds categoryIds to query
+   * @returns array of SubCategoryResultRow
+   */
+  public async querySubCategories(categoryIds: Id64String[]): Promise<SubCategoryResultRow[]> {
+    // consider splitting up categoryIds, as queries get slow with many many categoryids in them.
+    const maxCategoriesPerQuery = 200;
+    const result = [];
+    while (categoryIds.length !== 0) {
```

Comment:
Probably should let `IModelDb.query` worry about paging the results (it handles this for you).
If so, you can drop the need to call `categoryIds.length` and generalize the function to take an `Iterable<Id64String>`.

Reply:
If I do this, would I just do a for.. of loop and query each categoryId individually? Would that be any slower than having a larger where clause? Or should I build the entire where clause and then query?

Reply:
Entire where clause, otherwise ECSql can't handle paging for you.